code/game/objects/items/dice.dm

```diff
@@ -71,6 +71,62 @@
 		result = roll(sides)
 	update_appearance()
 
+/obj/item/dice/attack_self(mob/user)
+	diceroll(user, TRUE)
+
+/obj/item/dice/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	var/mob/thrown_by = thrownby?.resolve()
+	if(thrown_by)
+		diceroll(thrown_by)
+	return ..()
+
+/obj/item/dice/proc/diceroll(mob/user, in_hand=FALSE)
+	result = roll(sides)
+	if(rigged != DICE_NOT_RIGGED && result != rigged_value)
+		if(rigged == DICE_BASICALLY_RIGGED && prob(clamp(1/(sides - 1) * 100, 25, 80)))
+			result = rigged_value
+		else if(rigged == DICE_TOTALLY_RIGGED)
+			result = rigged_value
+
+	. = result
+
+	var/fake_result = roll(sides)//Daredevil isn't as good as he used to be
+	var/comment = ""
+	if(sides > MIN_SIDES_ALERT && result == 1)  // less comment spam
+		comment = "Ouch, bad luck."
+	if(sides == 20 && result == 20)
+		comment = "NAT 20!"  // maint wanted this hardcoded to nat20 don't blame me
+	update_appearance()
+	result = manipulate_result(result)
+	if(special_faces.len == sides)
+		comment = ""  // its not a number
+		result = special_faces[result]
+		if(!ISINTEGER(result))
+			comment = special_faces[result]  // should be a str now
+
+	if(in_hand) //Dice was rolled in someone's hand
+		user.visible_message(span_notice("[user] rolls [src]. It lands on [result]. [comment]"), \
+			span_notice("You roll [src]. It lands on [result]. [comment]"), \
+			span_hear("You hear [src] rolling, it sounds like a [fake_result]."))
+	else
+		visible_message(span_notice("[src] rolls to a stop, landing on [result]. [comment]"))
```

Comment:
```suggestion
		visible_message(span_notice("[src] rolls to a stop, landing on [result]. [comment]"))
		
	return .
```

just to make it abundantly clear what our dot operator is doing/the fact that we already set the dot operator earlier